spec/unit/puppet/provider/cs_colocation_crm_spec.rb

```diff
@@ -0,0 +1,140 @@
+require 'spec_helper'
+
+describe Puppet::Type.type(:cs_colocation).provider(:crm) do
+  before do
+    described_class.stubs(:command).with(:crm).returns 'crm'
+  end
+
+  context 'when getting instances' do
+    let :instances do
+
+      test_cib = <<-EOS
+        <cib>
+        <configuration>
+          <constraints>
+            <rsc_colocation id="first_with_second" rsc="first" score="INFINITY" with-rsc="second"/>
+          </constraints>
+        </configuration>
+        </cib>
+      EOS
+
+      described_class.expects(:block_until_ready).returns(nil)
+      if Puppet::PUPPETVERSION.to_f < 3.4
+        Puppet::Util::SUIDManager.expects(:run_and_capture).with(['crm', 'configure', 'show', 'xml']).at_least_once.returns([test_cib, 0])
+      else
+        Puppet::Util::Execution.expects(:execute).with(['crm', 'configure', 'show', 'xml']).at_least_once.returns(
+          Puppet::Util::Execution::ProcessOutput.new(test_cib, 0)
+        )
+      end
```

Comment:
I'm not happy about how this whole block mixes expectation setup and data creation. maybe split it into test_cib and instances properties and do the expectations in a before block?